sale_import_base/tests/test_sale_order_import.py

```diff
@@ -0,0 +1,218 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+import json
+
+from odoo.exceptions import ValidationError
+
+from .common_sale_order_import import SaleImportCase
+
+
+class TestSaleOrderImport(SaleImportCase):
+    def setUp(self):
+        super().setUp()
+
+    def test_invalid_json(self):
+        """ An invalid input will stop the job """
+        json_import = self.sale_data
+        del json_import["address_customer"]["street"]
+        with self.assertRaises(ValidationError):
+            self.importer_component.run(json.dumps(json_import))
+
+    def test_create_partner(self):
+        """ Base scenario: create partner """
+        json_import = self.sale_data
+        partner_count = self.env["res.partner"].search_count([])
+        new_sale_order = self.importer_component.run(json.dumps(json_import))
+        partner_count_after_import = self.env["res.partner"].search_count([])
+        self.assertEqual(partner_count_after_import, partner_count + 3)
+        binding_count = self.env["res.partner.binding"].search_count(
+            [
+                ("sale_channel_id", "=", self.sale_channel_ebay.id),
+                ("partner_id", "=", new_sale_order.partner_id.id),
+                ("external_id", "=", "ThomasJeanEbay"),
+            ]
+        )
+        self.assertEqual(binding_count, 1)
+
+    def test_import_existing_partner_match_external_id(self):
+        """ During import, if a partner is matched on external_id/channel
+        combination, his address is updated """
+        # Import once to create partner
+        json_import = self.sale_data
+        self.importer_component.run(json.dumps(json_import))
+        # New import that updates partner
+        del json_import["address_customer"]["email"]
+        json_import["address_customer"]["street"] = "new street"
+        json_import["payment"]["reference"] = "PMT-EXAMPLE-002"
+        new_sale_order = self.importer_component.run(json.dumps(json_import))
+        self.assertEqual(new_sale_order.partner_id.street, "new street")
+        binding_count = self.env["res.partner.binding"].search_count(
+            [
+                ("sale_channel_id", "=", self.sale_channel_ebay.id),
+                ("partner_id", "=", new_sale_order.partner_id.id),
+                ("external_id", "=", "ThomasJeanEbay"),
+            ]
+        )
+        self.assertEqual(binding_count, 1)
+
+    def test_import_existing_partner_match_email(self):
+        """ During import, if a partner is matched on email,
+        its address is updated """
+        # Import once to create partner
+        json_import = self.sale_data
+        self.importer_component.run(json.dumps(json_import))
+        # New import that updates partner
+        json_import["address_customer"]["external_id"] = "SomethingNew"
+        json_import["address_customer"]["street"] = "new street"
+        json_import["payment"]["reference"] = "PMT-EXAMPLE-002"
```

Comment:
too complex, I not sure we need to change this value, every line matter, less code, less complexity more readability